include/swift/Remote/MetadataReader.h

```diff
@@ -886,6 +886,77 @@ class MetadataReader {
     return resolver.swiftProtocol(Demangled);
   }
 
+  BuiltType readTypeFromShape(
+      RemoteAbsolutePointer shapeAddress, bool non_unique,
+      std::function<std::optional<std::vector<BuiltType>>(unsigned)> getArgs) {
+    StoredPointer addr = shapeAddress.getResolvedAddress().getAddressData();
+    // Non-unique shapes start with an offset to a cache variable.
+    if (non_unique)
+      addr += 4;
```

Comment:
@rjmccall Non-unique shapes have an extra field at the beginning:
https://github.com/swiftlang/swift/blob/a654df7caaff587aaf548fa5ef42b82ce35966bb/lib/IRGen/GenMeta.cpp#L7748

I'm adding the offset here, so the existing code remains unchanged, but maybe it would be better to teach readShape about this? It uses the address as a cache entry.

Reply:
To summarize an offline conversation: Shape symbols have and extra field, but all metadata pointers point directly to the shape data after that field, so this is the desired behavior.